api/cron-jobs/projects.js

```diff
@@ -129,51 +153,88 @@ function formatProjects(projects, resources) {
-      const projectsFilteredAndFormatted = module.exports.formatProjects(projectsFiltered, resourcesFiltered);
-
-      console.log(
-        `🛈 Found ${projectsFilteredAndFormatted.length} projects matching resources, ${resourcesFiltered.length} resources matching projects`,
-      );
-
-      resolve({
-        projects: projectsFilteredAndFormatted,
-        resources: resourcesFiltered,
-      });
+      try {
+        const itArray = data[0];
+        const resArray = data[1];
+
+        if (data[0] === undefined || data[1] === undefined)
+          throw new Error(
+            'Initial triage data or resource data returned as undefined'
+          );
+
+        const projectsFiltered =
+          module.exports.filterProjectsConnectedWithResources(
+            itArray,
+            resArray
+          );
+        const resourcesFiltered =
+          module.exports.filterResourcesConnectedWithProjects(
+            itArray,
+            resArray
+          );
+        const projectsFilteredAndFormatted = module.exports.formatProjects(
+          projectsFiltered,
+          resourcesFiltered
+        );
+
+        console.log(
+          `🛈 Found ${projectsFilteredAndFormatted.length} projects matching resources, ${resourcesFiltered.length} resources matching projects`
+        );
+
+        resolve({
+          projects: projectsFilteredAndFormatted,
+          resources: resourcesFiltered,
+        });
+      } catch (e) {
+        console.log(e);
+        return;
+      }
     });
   });
 }
 
 async function getInitialTriageProjectsFromJira(startAt, itArray) {
   const itJqlQuery = encodeURIComponent(
-    `project=${projectJiraBoardName} AND status="${volunteerSearch}" OR status="${volunteerIntroduction}" OR status="${activityUnderway}"`,
-  );
-  const jiraIt = await axios.get(
-    `https://sta2020.atlassian.net/rest/api/2/search?jql=${itJqlQuery}&startAt=${startAt}&maxResults=1000`,
-    {
-      headers: {
-        Authorization: `Basic ${Buffer.from(
-          // below use email address you used for jira and generate token from jira
-          `${email}:${api_key}`,
-        ).toString('base64')}`,
-        Accept: 'application/json',
-      },
-    },
+    `project=${projectJiraBoardName} AND status='${volunteerSearch}' OR status='${volunteerIntroduction}' OR status='${activityUnderway}'`
   );
 
+  let jiraIt;
+  try {
+    jiraIt = await axios.get(
+      `https://sta2020.atlassian.net/rest/api/2/search?jql=${itJqlQuery}&startAt=${startAt}&maxResults=1000`,
+      {
+        headers: {
+          Authorization: `Basic ${Buffer.from(
+            // below use email address you used for jira and generate token from jira
+            `${email}:${api_key}`
+          ).toString('base64')}`,
+          Accept: 'application/json',
+        },
+      }
+    );
+    if (!jiraIt?.data?.issues?.length)
+      throw new Error('Jira returned no initial triage projects');
+  } catch (error) {
+    console.error(error);
```

Comment:
This is personal preference but I find it useful to return _what_ we were doing when the unexpected error was thrown. Something like "While accessing Axios API {error} was caught".

Reply:
I've added a [new commit](https://github.com/Scottish-Tech-Army/Volunteer-app/pull/136/commits/7d2d8a4d7692282bc3c5a45da8fd5296fdad8918) on this with what you're suggesting, am I along the right lines? What you say makes sense.

Reply:
That looks useful, thanks @kelsiesmurphy -- although it would be good if those messages could be a little more specific to help someone identify where it happened e.g. 'While getting initial triage projects from Jira' rather than a more generic 'While accessing Axios'.

Also bear in mind the `error` variable could be an object (e.g. if it was an error thrown by Axios itself) so I wouldn't put it into the string as-is: you could add it as a second argument of `console.error` (or you could `JSON.stringify` it and add it to the end of the string)

Reply:
Good point, will update with a commit just now.

Reply:
[Linked commit](https://github.com/Scottish-Tech-Army/Volunteer-app/pull/136/commits/e764840e28d83ef996d2b9790a4c472c7f2e4fea)